contracts/feeds/PriceFeedV1PoolOracle.sol

```diff
@@ -0,0 +1,111 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+import "./PriceFeeds.sol";
+import "./IV1PoolOracle.sol";
+import "../openzeppelin/Ownable.sol";
+import "../openzeppelin/Address.sol";
+import "./IPriceFeeds.sol";
+
+/**
+ * @notice The Price Feed V1 Pool Oracle contract.
+ *
+ * This contract implements V1 Pool Oracle query functionality,
+ * getting the price and the last timestamp from an external oracle contract.
+ * */
+contract PriceFeedV1PoolOracle is IPriceFeedsExt, Ownable {
+	/* Storage */
+
+	address public v1PoolOracleAddress;
+	address public rBTCAddress;
+	address public docAddress;
+
+	/* Events */
+	event SetV1PoolOracleAddress(address indexed v1PoolOracleAddress, address changerAddress);
+	event SetRBTCAddress(address indexed rBTCAddress, address changerAddress);
+	event SetDOCAddress(address indexed docAddress, address changerAddress);
+
+	/* Functions */
+
+	/**
+	 * @notice Initialize a new V1 Pool Oracle.
+	 *
+	 * @param _v1PoolOracleAddress The V1 Pool Oracle address.
+	 * */
+	constructor(
+		address _v1PoolOracleAddress,
+		address _rBTCAddress,
+		address _docAddress
+	) public {
+		setV1PoolOracleAddress(_v1PoolOracleAddress);
+		setRBTCAddress(_rBTCAddress);
+		setDOCAddress(_docAddress);
+	}
+
+	/**
+	 * @notice Get the oracle price.
+	 * @return The price from Oracle.
+	 * */
+	function latestAnswer() external view returns (uint256) {
+		require(rBTCAddress != address(0), "rBTC address has not been set");
+		require(docAddress != address(0), "DOC address has not been set");
+
+		IV1PoolOracle _v1PoolOracle = IV1PoolOracle(v1PoolOracleAddress);
+		// Need to check, if the requested asset is BTC
+		address liquidityPool = _v1PoolOracle.liquidityPool();
+		require(
+			ILiquidityPoolV1Converter(liquidityPool).reserveTokens(0) != rBTCAddress ||
+				ILiquidityPoolV1Converter(liquidityPool).reserveTokens(1) != rBTCAddress,
```

Comment:
Yes correct.
In oracle v1 pool, the price is based on BTC. So, oracleV1PriceFeed, we will convert the price that we get from the oracle v1 pool to USD (because our other price feed is based on USD).
In order to convert the price from BTC to USD, oracleV1PriceFeed will query the wRBTC priceFeed. That's why we need to prevent wRbtc priceFeed to be set to oracleV1PriceFeed, otherwise, this will cause infinite loop.

Reply:
it will return true only if both tokens are not WRBTC which i believe is always true.
only false if both  are wrbtc
are you sure it is what you are checking?

Reply:
Yes, this is what we are trying to check (make sure WRBTC oracle priceFeed is not being registered into this v1 priceFeed)